### PR TITLE
Added comments to bark server

### PR DESCRIPTION
**Pull request by bhagatvansh**
Added documentation in the form of comments for bark server.

Open to suggestions/feedbacks on improvements needed.

### CODE REVIEW

File: cmd/server/main.go [hunk 1]
```diff
@@ -31,21 +31,38 @@ func Init() string {
...
 	r.GET("/hello/{name}", Hello)
+
+	// Bark client contains the logic which decides which out of the two (single/multiple) insertion endpoints is called.
+	// This endpoint sends single log entry at a time to the DB.
 	r.POST("/insertSingle", controllers.SendSingleToChannel)
+	// This endpoint handles the batch insertion of logs to the DB.
 	r.POST("/insertMultiple", controllers.SendMultipleToChannel)
+
+	// This endpoint is responsible to initiate server shut down, following which Bark server will not process any new incoming requests.
+	// It will, however, shut down after it has completely sent all the logs received up till that point of time.
```
Comment (vaibhav-kaushal) [inline cmd/server/main.go:48]: We should use the words "has completely saved all logs received till that point to the database" - makes it more clear. The word "sent" makes me feel like the logs are being further forwarded to some other place.

File: services/ingestion/read_from_channel.go [hunk 2]
```diff
@@ -8,6 +8,9 @@ import (
 	"github.com/techrail/bark/models"
 )
 
+// InsertSingle and InsertMultiple functions have a common functionality - pushing the logs into the LogChannel.
+// The channel capacities are checked beforehand. In case, the channels are full, that message is conveyed.
+// Each log entry's values are validated and then sent to the LogChannel.
```
Comment (vaibhav-kaushal) [inline services/ingestion/read_from_channel.go:13]: We should still write a single line comment for both functions, for the sake of uniformity across the codebase.

**Review by vaibhav-kaushal — CHANGES_REQUESTED** [review]
Very small changes required.

File: resources/dbPoolConfig.go [hunk 3]
```diff
@@ -13,6 +13,10 @@ import (
 	"github.com/jackc/pgx/v5/pgxpool"
 )
 
+// Config retrieves the postgres DB connection url from environment variable named `BARK_DATABASE_URL`.
+// It then parses the connection url and checks for empty and invalid urls, if found, it logs the error and connection is not made.
+// If it passes the checks, the connection is established and function returns object of pgxpool config.
+
```
Comment (itishrishikesh) [inline resources/dbPoolConfig.go:19]: I think this new line can be removed.

**Review by itishrishikesh — APPROVED** [review]
Thank you @bhagatvansh !! LGTM.

Just added a very minor/trivial comment, which can be added later as well.

File: resources/dbPoolConfig.go (hunk 3 above)
Comment (bhagatvansh) [reply resources/dbPoolConfig.go]: Removed the new line.

File: services/ingestion/read_from_channel.go (hunk 2 above)
Comment (bhagatvansh) [reply services/ingestion/read_from_channel.go:13]: Added comments for both functions.

File: cmd/server/main.go (hunk 1 above)
Comment (bhagatvansh) [reply cmd/server/main.go]: Worded the comment in a more comprehensible way as suggested.

**Pull request merged**